agent/local/state.go

```diff
@@ -275,20 +284,21 @@ func (l *State) addServiceLocked(service *structs.NodeService, token string) err
 	}
 
 	l.setServiceStateLocked(&ServiceState{
-		Service: service,
-		Token:   token,
+		Service:          service,
+		Token:            token,
+		IsLocallyDefined: isLocal,
 	})
 	return nil
 }
 
 // AddServiceWithChecks adds a service entry and its checks to the local state atomically
 // This entry is persistent and the agent will make a best effort to
 // ensure it is registered
-func (l *State) AddServiceWithChecks(service *structs.NodeService, checks []*structs.HealthCheck, token string) error {
+func (l *State) AddServiceWithChecks(service *structs.NodeService, checks []*structs.HealthCheck, token string, isLocal bool) error {
```

Comment:
A todo is to figure out if registering checks should also use the config_file_registration token. I imagine probably so, but I haven't thought it through yet.